tests/unit/models/Component.spec.js

```diff
@@ -49,44 +51,213 @@ describe('Test class: Component', () => {
+                type: 'Reference',
+                containerRef: 'container',
+              }),
+            ],
+          }),
+        });
+        component.setContainerAttribute(new Component({
+          id: 'containerId',
+          definition: new ComponentDefinition({
+            type: 'container',
+          }),
+        }));
+
+        expect(component.attributes.length).toEqual(1);
+        expect(component.attributes[0].name).toEqual('container');
+        expect(component.attributes[0].value).toEqual('containerId');
+        expect(component.attributes[0].definition).not.toBeNull();
       });
-      it('Check not passing child if parentType and type not equal', () => {
-        const componentDefinition = new ComponentDefinition({ isContainer: true, type: 'parent1' });
-        const parent = new Component({ id: '1', definition: componentDefinition });
-        const child = new Component({ id: '2', definition: { parentTypes: ['parent2'] } });
-        parent.addChild(child);
-        expect(parent.children.length).toEqual(0);
+
+      it('should update attribute if container attribute already exits', () => {
+        const component = new Component({
+          attributes: [
+            new ComponentAttribute({
+              name: 'container',
+              definition: new ComponentAttributeDefinition({
+                name: 'container',
+                type: 'Reference',
+                containerRef: 'container',
+              }),
+              value: 'test',
+            }),
+          ],
+          definition: new ComponentDefinition({
+            definedAttributes: [
+              new ComponentAttributeDefinition({
+                name: 'container',
+                type: 'Reference',
+                containerRef: 'container',
+              }),
+            ],
+          }),
+        });
+        component.setContainerAttribute(new Component({
+          id: 'containerId',
+          definition: new ComponentDefinition({
+            type: 'container',
+          }),
+        }));
+
+        expect(component.attributes.length).toEqual(1);
+        expect(component.attributes[0].name).toEqual('container');
+        expect(component.attributes[0].value).toEqual('containerId');
+        expect(component.attributes[0].definition).not.toBeNull();
       });
-      it('Check not passing child if it already exists in children', () => {
-        const componentDefinition = new ComponentDefinition({ isContainer: true, type: 'parent1' });
-        const parent = new Component({ id: '1', definition: componentDefinition });
-        const child = new Component({ id: '2', definition: { parentTypes: ['parent1'] } });
-        parent.addChild(child);
-        expect(parent.children.length).toEqual(1);
-        parent.addChild(child);
-        expect(parent.children.length).toEqual(1);
+    });
+
+    describe('Test method: unsetContainerAttribute', () => {
+      it('Should remove attribute if exists', () => {
```

Comment:
`attribute if exists`
-> `attribute if it exists`
-> `existing attribute`

Reply:
done